### DISCUSSION TITLE
The program has been running without a result

### DISCUSSION THREAD

**Discussion by wendy444444** [yalmip/YALMIP#1212] [General discussion] [answered]
Hello, sir. I added the following section to the program that could produce results, `pritim` is the variable that could have figured out before. And then the program kept running without results.
```
xz =0.5;
thva = 10^( -3 );
PGT = sdpvar( 1, 10, 'full' );
GGT = sdpvar( 1, 5, 'full' );
         C = [ C, (implies(t<=pritim(1),PGT(t)==0)):'PGT==xz*GGT' ];
         C = [ C, (implies(t-sum(pritim)>=thva,PGT(t)==xz*GGT(time))):'PGT==xz*GGT' ];
    for t = 1:10
        for q=2:5
         C = [ C, (implies((sum(pritim(1:q-1))<=t)&(t<=sum(pritim(1:q))),PGT(t)==xz*GGT(q-1))):'PGT==xz*GGT' ];
        end
    end
```
I guess the problem is "<=>", but I don't know how to solve it. Could you help me, please?

**Comment by jdbancal** [comment]
Where is your call to `optimize`?

**Reply by wendy444444** [reply]
Thank you. The problem has been solved

**Comment by johanlofberg** [comment, marked as answer]
First, I assume you haven't turned off display, so you see what is going on.

The model you describe will be a huge mixed-integer model (from implies), so I guess the solver (which you haven't revealed to us) simply isn't capable of solving the problem

If pritim is a constant as you seem to suggest, you should not use implies, then you shouldn't use implies as it just complicates things, and the stall must come from somewhere else, impossible to say without logs

**Reply by wendy444444** [reply]
My solver is gurobi, and my objective is that the variable is the minimum. The variable and PGT and GGT, have an indirect relationship with the target. ‘pritim’ is similar to PGT and GGT, so it is not a variable, and I use implies. After modifying the constraints, I got the result. Thank you again.

**Reply by johanlofberg** [reply]
So you mean pritim is an expression built from variables.

You mean the problem is solved now?

**Reply by wendy444444** [reply]
Yes, as you understand. The problem was with the constraints, which have now been resolved. Thank you very much for your help.
